engine/gfx/sprite_palettes.asm

```diff
@@ -1,3 +1,53 @@
+LoadWeatherPalNoApply:
+	ld hl, wPalFlags
+	set NO_DYN_PAL_APPLY_ONCE_F, [hl]
+LoadWeatherPal::
+	ld a, [wCurrentWeather]
+	and a
+	ret z ; no weather
+	dec a
```

Comment:
Some `assert`s here as to the weather constant values would be appropriate. (pokecrystal recently added more of those when using this `dec a :: jr z` optimization.)